### ISSUE TITLE
Convert build scripts to gh actions

### ISSUE THREAD

**Issue by richard-cox** [rancher/dashboard#10287]
- Convert contents of `.drone.yaml` in both rancher/dashboard and rancher/ui to gh actions
- This needs to happen before June

Main issue is uploading to Google Cloud Storage. So a replacement for https://plugins.drone.io/plugins/gcs

```
- name: upload
  pull: default
  image: plugins/gcs
  settings:
    acl:
    - allUsers:READER
    cache_control: "no-cache,must-revalidate"
    source: dist/${DRONE_BRANCH}.tar.gz
    target: releases.rancher.com/dashboard/${DRONE_BRANCH}.tar.gz
    token:
      from_secret: google_auth_key
```

with something like https://github.com/google-github-actions/upload-cloud-storage.

Once we confirm the replacement and the auth required, we need to raise with eio

**Comment by richard-cox** [comment]
closing in favour of newer more targeted issues (see links above)
